workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentDataGenerator.tt

```diff
@@ -123,6 +137,22 @@ namespace <#= qualifiedNamespace #>
                 return update;
             }
 
+            public static <#= componentNamespace #>.Snapshot DeserializeSnapshot(global::Improbable.Worker.Core.SchemaObject obj, global::Unity.Entities.World world)
+            {
+                var component = new <#= componentNamespace #>.Snapshot();
+
+<# foreach (var fieldDetails in fieldDetailsList) { #>
+<# if (!fieldDetails.IsSingular) { #>
+                component.<#= fieldDetails.PascalCaseName#> = new <#= fieldDetails.Type #>();
+<# } #>
+                {
+                    <#= fieldDetails.GetDeserializeString("component." + fieldDetails.PascalCaseName, "obj", 5) #>
```

Comment:
Instead of the if + the GetDeserialiseString - why not add a `GetDeserialiseSnapshotString` which handles both?

Reply:
Actually you don't even need the if part. The DeserialiseString includes instantiating the field. Look at some of the non-blittable generated code for examples

Reply:
You're right. I'm always caught out by the "Allocate" function not allocating anything. It makes it look like it's doing something very different.

Also that string looks like it's doing something weird for non-blittable types. I'm seeing multiple `=` on the same line. Will refactor that.